Team Sync: Add group mapping to support team sync in the Generic OAuth provider (#36307)
grafana/grafana: docs/sources/auth/generic-oauth.md

Check for the presence of a role using the [JMESPath](http://jmespath.org/examples.html) specified via the `role_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the UserInfo endpoint specified via the `api_url` configuration option. The result after evaluating the `role_attribute_path` JMESPath expression needs to be a valid Grafana role, i.e. `Viewer`, `Editor` or `Admin`.

> Grafana will also attempt to do team mapping through OAuth as described below.
> 
> Check for the presence of groups using the [JMESPath](http://jmespath.org/examples.html) specified via the `groups_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the UserInfo endpoint specified via the `api_url` configuration option. The result after evaluating the `groups_attribute_path` JMESPath expression needs to return a comma delimited string with group_id's. 

Reviewer: ```suggestion
Grafana also attempts to map teams through OAuth as described below.

Check for the presence of groups using the [JMESPath](http://jmespath.org/examples.html) specified via the `groups_attribute_path` configuration option. The JSON used for the path lookup is the HTTP response obtained from querying the UserInfo endpoint specified via the `api_url` configuration option. After evaluating the `groups_attribute_path` JMESPath expression, the result needs to return a comma delimited string with group_id's. 
```

Config:
```bash
role_attribute_path = contains(info.roles[*], 'admin') && 'Admin' || contains(info.roles[*], 'editor') && 'Editor' || 'Viewer'
```

### Groups mapping

>  Available in Grafana Enterprise v8.1 and later versions.

With Team Sync you can map your Generic OAuth groups to teams in Grafana so that the users are automatically added to the correct teams.

Generic OAuth groups can be referenced by group ID, like `8bab1c86-8fba-33e5-2089-1d1c80ec267d` or `myteam`.

[Learn more about Team Sync]({{< relref "team-sync.md" >}})

Config:

```bash

> groups_attribute_path = info.groups.join(',',[*])  

Reviewer 2: Here's where it would make more sense to me if this was just `groups_attribute_path = info.groups`, i.e. a string array

```bash
role_attribute_path = contains(info.groups[*], 'admin') && 'Admin' || contains(info.groups[*], 'editor') && 'Editor' || 'Viewer'
```

### Groups mapping

> >  Only available in Grafana Enterprise v8.1+
> 
> With Team Sync you can map your Generic OAuth groups to teams in Grafana so that your users will automatically be added to the correct teams.

Reviewer: ```suggestion
>  Available in Grafana Enterprise v8.1 and later versions.

With Team Sync you can map your Generic OAuth groups to teams in Grafana so that the users are automatically added to the correct teams.
```

```bash
role_attribute_path = contains(info.groups[*], 'admin') && 'Admin' || contains(info.groups[*], 'editor') && 'Editor' || 'Viewer'
```

### Groups mapping

> >  Only available in Grafana Enterprise v6.7+

Reviewer 3: ```suggestion
>  Only available in Grafana Enterprise v8.1+
```
